src/Stratis.Bitcoin.Features.Wallet/WalletManager.cs

```diff
@@ -311,18 +311,19 @@ public Mnemonic CreateWallet(string password, string name, string passphrase, Mn
         }
 
         /// <inheritdoc />
-        public string SignMessage(string password, string walletName, string externalAddress, string message)
+        public string SignMessage(string password, string walletName, string accountName, string externalAddress, string message)
         {
             Guard.NotEmpty(password, nameof(password));
             Guard.NotEmpty(walletName, nameof(walletName));
+            Guard.NotEmpty(accountName, nameof(accountName));
             Guard.NotEmpty(message, nameof(message));
             Guard.NotEmpty(externalAddress, nameof(externalAddress));
 
             // Get wallet
             Wallet wallet = this.GetWalletByName(walletName);
 
             // Sign the message.
-            HdAddress hdAddress = wallet.GetAddress(externalAddress);
+            HdAddress hdAddress = wallet.GetAddress(externalAddress, account => account.Name.Equals(accountName));
```

Comment:
Can you elaborate a bit more on this logic?
why the need for account name here? address should anyway be unique no?

Reply:
I think the initial thought was to be able to pass multiple account names to search for the address, but you are right, I will change this to pass the single value through.

Reply:
Looking at both GetAddress and GetAllAddresses methods, they require an account name filter, so this is good.